src/actions/guides/database/validating-saving-deleting.cr

```diff
@@ -453,75 +448,79 @@ class Guides::Database::ValidatingSavingDeleting < GuideAction
     * `:save` if required for update and create, but not when calling `new`
 
     ```crystal
-    class CommentForm < Comment::BaseForm
+    class SaveComment < Comment::SaveOperation
       needs author : User, on: :create # can also be `:update`, `:save`
 
-      def prepare
+      before_save prepare_comment
+
+      def prepare_comment
         author.try do |user|
           authored_by_id.value = user.id
         end
       end
     end
     ```
 
-    > Note that `author` is not required when calling `UserForm.new` when using
+    > Note that `author` is not required when calling `SaveUser.new` when using
     the `on` option. This means `author` can be `nil` in the form. That's why we
-    needed to use `try` in the `prepare` method.
+    needed to use `try` in the `prepare_comment` method.
 
     ```crystal
     # You must pass an author when creating
-    CommentForm.create(params, author: a_user) do |form, user|
+    SaveComment.create(params, author: a_user) do |operation, user|
       # do something
     end
 
     # But you can't when you are updating
-    CommentForm.update(comment, params) do |form, user|
+    SaveComment.update(comment, params) do |operation, user|
       # do something
     end
 
-    # You also can't pass it in when instantiating a new CommentForm
-    CommentForm.new
+    # You also can't pass it in when instantiating a new SaveComment
+    SaveComment.new
     ```
 
-    > **When should I use `on`?** If you are building an server rendered HTML app
+    > **When should I use `on`?** If you are building a server rendered HTML app,
     then you will almost always wants to use `on :save|:update|:create` because
-    you will call `MyForm.new` without the needs. If you are building a JSON API
+    you will call `SaveComment.new` without the needs. If you are building a JSON API
     you may want to omit the `on` option since you rarely use `.new`. If you omit
     `on` then you don't need to worry about the value ever being `nil`, which can
     make your program more reliable and easier to understand.
 
-    ## Virtual fields
+    ## Non-model attributes
```

Comment:
How about “non-database column attributes”?

Reply:
That feels a bit long 😅 Maybe "Virtual attributes"? lol I don't know. I'll change it for now, but if you think of something shorter, let me know.